controllers/update_pods.go

```diff
@@ -72,11 +72,47 @@ func (updatePods) reconcile(ctx context.Context, r *FoundationDBClusterReconcile
 	return deletePodsForUpdates(ctx, r, cluster, updates, logger, status)
 }
 
+// getFaultDomainsWithUnavailablePods returns a map of fault domains with unavailable Pods. The map has the fault domain as key and the value is not used.
+func getFaultDomainsWithUnavailablePods(ctx context.Context, reconciler *FoundationDBClusterReconciler, cluster *fdbv1beta2.FoundationDBCluster) map[fdbv1beta2.FaultDomain]fdbv1beta2.None {
+	faultDomainsWithUnavailablePods := make(map[fdbv1beta2.FaultDomain]fdbv1beta2.None)
+
+	for _, processGroup := range cluster.Status.ProcessGroups {
+		unavailable := false
+		// If the Pod is pending, we count it as unavailable.
+		if processGroup.GetConditionTime(fdbv1beta2.PodPending) != nil {
+			unavailable = true
+		}
+		// If the Pods is initializing, we count it as unavailable.
+		if processGroup.GetConditionTime(fdbv1beta2.PodInitializing) != nil {
+			unavailable = true
+		}
+		// If the Pod is running with failed containers, we count it as unavailable.
+		if processGroup.GetConditionTime(fdbv1beta2.PodFailing) != nil {
+			unavailable = true
+		}
+		pod, err := reconciler.PodLifecycleManager.GetPod(ctx, reconciler, cluster, processGroup.GetPodName(cluster))
+		// If a Pod is not found consider it as unavailable.
```

Comment:
We probably should log the error if the error is a different one than `..not found`. I would also prefer to return any errors that are not `.. not found` otherwise we might be acting on the wrong signal.